.github/workflows/version-or-publish.yml

```diff
@@ -0,0 +1,85 @@
+name: Version or Publish Package
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_tag:
+        required: true
+        description: The release version
+        type: string
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+permissions:
+  id-token: write
+  contents: write
+  attestations: write
+  actions: read
+
+jobs:
+  provenance:
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_nodejs_slsa3.yml@v1.10.0
+    with:
+      run-scripts: "install-pnpm, install-deps, style, nx-build-skip-cache, nx-test-skip-cache, skip-lib-ignore"
+      node-version: "20.11.1"
+
+  release:
+    needs: [provenance]
+    name: Version or Publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Downlod Artifacts 
+        uses: actions/download-artifact@v3
+
+      - name: Checkout Repo
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.5.4
+        with:
+          token: ${{ secrets.SAI_PAT }}
+
+      - name: Prepare pre-requisites
+        uses: ./.github/actions/prepare
+
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@01dd5d3ca463c7f10f7f4f7b4f177225ac661ee4
+        with:
+          gpg_private_key: ${{ secrets.SVC_GPG_KEY }}
+          passphrase: ${{ secrets.SVC_GPG_PASSPHRASE }}
+          git_config_global: true
+          git_tag_gpgsign: true
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          setupGitUser: false
+          version: pnpm ci:version
+          title: "Update the version packages"
+          publish: pnpm release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
wondering how does this work? since we are trying to update manually the root package.json and then pushing those changes to the changeset pr raised by the changeset bot.

Reply:
> a PR won't be created but npm publish will happen. The reverse is true. 

Assuming this is PR will be create and npm publish will happen after